arangod/Agency/Constituent.cpp

```diff
@@ -681,7 +673,7 @@ void Constituent::run() {
         int64_t randTimeout = RandomGenerator::interval(a, b);
         int64_t randWait = randTimeout;
 
-        {
+       {
           MUTEX_LOCKER(guard, _castLock);
```

Comment:
seems like wrong lock.  _heartBeatMutex seems defined in Constituent.h as proper lock.

Reply:
Actually, I believe no lock is needed whatsoever. We are only accessing `_lastHeartbeatSeen` which is a `std::atomic<double>` and this is OK. The map `_lastHeartbeatSent` is protected by `_heartBeatMutex`. But we do not need `_castLock` or now `_termVoteLock` at all here. Changed.